azuread/resource_service_principal_password.go

```diff
@@ -58,6 +58,13 @@ func resourceServicePrincipalPasswordCreate(d *schema.ResourceData, meta interfa
 
 	d.SetId(id.String())
 
+	_, err = graph.WaitForReplication(func() (interface{}, error) {
+		return client.Get(ctx, id.ObjectId)
```

Comment:
So i think this should be:
```suggestion
		return client.ListPasswordCredentials(ctx, id.ObjectId)
```

as that is how we are checking if the credentials exist?